fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/rendering/v1/FabricSkyPropertyBuilder.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.rendering.v1;
+
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+
+import com.mojang.datafixers.util.Function3;
+
+import net.minecraft.client.render.SkyProperties;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Vec3d;
+
+public final class FabricSkyPropertyBuilder {
+	private Function3<float[], Float, Float, float[]> fogColorOverride = (color, skyAngle, tickDelta) -> {
+		float g = MathHelper.cos(skyAngle * 6.2831855F) - 0.0F;
+
+		if (g >= -0.4F && g <= 0.4F) {
+			float i = (g - -0.0F) / 0.4F * 0.5F + 0.5F;
+			float j = 1.0F - (1.0F - MathHelper.sin(i * 3.1415927F)) * 0.99F;
+			j *= j;
+			color[0] = i * 0.3F + 0.7F;
+			color[1] = i * i * 0.7F + 0.2F;
+			color[2] = i * i * 0.0F + 0.2F;
+			color[3] = j;
+			return color;
+		} else {
+			return null;
+		}
+	};
+
+	private float cloudsHeight = 128.0F;
+	private boolean alternateSkyColor = true;
+
+	private BiFunction<Vec3d, Float, Vec3d> adjustFogColor = (color, sunHeight) -> color.multiply((sunHeight * 0.94F + 0.06F), sunHeight * 0.94F + 0.06F, (sunHeight * 0.91F + 0.09F));
+	private BiPredicate<Integer, Integer> useThickFog = (camX, camY) -> false;
```

Comment:
I recommend dedicated functional interfaces that have their parameters named for clarity.